erp-server/src/main/java/com/ttt/erp/service/RecoveryService.java

```diff
@@ -48,12 +48,13 @@ public Boolean validateRecoveryQuestions(String username, final List<RecoveryQue
         return result.size() == recoveryQuestions.size();
     }
 
-    public Boolean updateUsersPassword(String username, final String password) {
+    public Boolean updateUsersPassword(String username, String password) {
         try {
             UserAccount existing = this.userAccountRepository.findByUsername(username);
             UserAccount modified = existing;
```

Comment:
```modified``` and ```existing``` are referencing the same object. That's why when you say ```modified.setPassword(password)```, the password on ```existing``` gets changed too and nothing gets logged. If objects are the same then there's of course no diff between them. To demonstrate, I pulled down your code and ran through the password reset.

My code change to print results looked like:
```Java
            System.out.println(modified.toString());
            System.out.println(existing.toString());
            modified.setPassword(password);
            System.out.println(modified.toString());
            System.out.println(existing.toString());
```

And in the console, it looked like:
```
UserAccount{id=2, email='demo@demo.com', username='foo', password='demo2', signature='d123c23f-1801-4002-91c0-d4cb43c4bdf0.png', isAdmin=false', isEnabled=true}
UserAccount{id=2, email='demo@demo.com', username='foo', password='demo2', signature='d123c23f-1801-4002-91c0-d4cb43c4bdf0.png', isAdmin=false', isEnabled=true}
UserAccount{id=2, email='demo@demo.com', username='foo', password='demo', signature='d123c23f-1801-4002-91c0-d4cb43c4bdf0.png', isAdmin=false', isEnabled=true}
UserAccount{id=2, email='demo@demo.com', username='foo', password='demo', signature='d123c23f-1801-4002-91c0-d4cb43c4bdf0.png', isAdmin=false', isEnabled=true}```

So confirmed that's the root cause.